### PR TITLE
Add schemas for Lifecycle & AccessControl Policies

### PR DESCRIPTION
**Pull request by mithmatt**
### Describe your changes :

I worked on the following as a first step towards defining lifecycle policies #950 

    Add schemas for Lifecycle & AccessControl Policies
    - Define Storage Class
    - Define Lifecycle Rule, DeleteAction and MoveAction
    - Define Rule and TagBased AccessControl

For Lifecycle actions, refer table [here](https://docs.google.com/document/d/1svyYBpSNBq3XUhALNCOuS-WTYtHNE-NMnghwszDe2G8/edit#heading=h.d2vjhtbaw370)

#
### Type of change :

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

Backend: @sureshms @harshach

### CODE REVIEW

File: catalog-rest-service/src/main/resources/json/schema/entity/services/storageService.json [hunk 1]
```diff
@@ -4,28 +4,6 @@
   "title": "Storage Service",
   "description": "This schema defines the Storage Service entity, such as S3, GCS, HDFS.",
   "type": "object",
-  "definitions": {
-    "storageServiceType": {
```
Comment (mithmatt) [inline catalog-rest-service/src/main/resources/json/schema/entity/services/storageService.json:8]: Moving this to type

File: catalog-rest-service/src/main/resources/json/schema/type/storage.json [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+          "name": "S3"
+        },
+        {
+          "name": "GCS"
+        },
+        {
+          "name": "HDFS"
+        },
+        {
+          "name": "ABFS"
```
Comment (mithmatt) [inline catalog-rest-service/src/main/resources/json/schema/type/storage.json:29]: Added new enum ABFS (at the end)

File: catalog-rest-service/src/main/resources/json/schema/type/storage.json [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        }
+      ]
+    },
+    "storageClassType": {
+      "description": "Type of storage class offered by the Storage Service",
+      "type": "object",
+      "javaType": "org.openmetadata.catalog.type.StorageClassType",
+      "enum": [
```
Comment (mithmatt) [inline catalog-rest-service/src/main/resources/json/schema/type/storage.json:37]: I'm not fully satisfied with defining these here.

Alternative approach:
Make storage class as a string. 

In data (json/schema/data/storageClasses/*.json) define all combinations of storage Classes

json/schema/data/storageClasses/gcs/nearline.json
```
{
  "storageServiceType": "GCS",
  "storageClassType": "Nearline"
}
```

@sureshms @harshach Any thoughts on this one?

File: catalog-rest-service/src/main/resources/json/schema/entity/services/storageService.json (hunk 1 above)
Comment (mithmatt) [reply catalog-rest-service/src/main/resources/json/schema/entity/services/storageService.json:8]: cc: @amiorin

Comment (amiorin) [reply catalog-rest-service/src/main/resources/json/schema/entity/services/storageService.json:8]: 👍

File: catalog-rest-service/src/main/resources/json/schema/type/storage.json (hunk 3 above)
Comment (amiorin) [reply catalog-rest-service/src/main/resources/json/schema/type/storage.json:37]: The storage class is per object in AWS while the storage service type is per group of buckets. I would use a copy of the AWS Java SDK enum. https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/s3/model/StorageClass.html

Comment (mithmatt) [reply catalog-rest-service/src/main/resources/json/schema/type/storage.json:37]: This is a great suggestion.

GCS
https://googleapis.dev/java/google-cloud-storage/latest/index.html

Azure
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/AccessTier.java

**Comment by harshach** [comment]
@amiorin can you review and see if we can merge this.

**Comment by mithmatt** [comment]
Thanks for merging the changes on this one.

**Pull request merged**